homework-submission/mobx-week1/src/stores/TodoStore.js

```diff
@@ -0,0 +1,40 @@
+import { observable, action, computed } from "mobx";
+// import moment from "moment";
+
+class TodoStore {
+  @observable todos = [];
+  @observable editing = false;
+
+  @action addTodo = todo => {
+    if (todo.description.length !== 0 && todo.deadline.length !== 0)
+      this.todos.push(todo);
+  };
+  @action onCheck = (e, id) => {
+    this.todos = this.todos.map(todo => {
```

Comment:
I don't see any use of `.map` here more than the iteration part, while we are looping through all the elements and mutating by there value, we are mutating them directly and not by reference.

with that using `.forEach` is a bit more reasonable, same as the next method.